### DISCUSSION TITLE
No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp() (app/no-app)

### DISCUSSION THREAD

**Discussion by stormfar** [vercel/next.js#11351] [Help] [answered]
I'm getting that error when calling firebase in `getInitialProps` or `getStaticProps`.

```
Page.getInitialProps = async ({ req }) => {
const db = await firebase.firestore(); // firebase is not ready yet
...
```

The issue seems to be that whatever call is being made is happening before `initializeApp()` is complete. In the [with-firebase](https://github.com/zeit/next.js/blob/canary/examples/with-firebase/firebase/clientApp.js), firebase is just configured and exported. So I guess it needs to loaded async somehow. 

Anyone have a solution for this?

**Comment by mikxyas** [comment, marked as answer]
I am facing the same issue

**Reply by brunocrosier** [reply]
You want something like this:

```javascript
// firebase.js

import fb from "firebase/app"
export const firebase = !fb.apps.length ? fb.initializeApp(firebaseConfig) : fb.app()

// somecomponent.js
import {firebase} from "../firebase.js"

// do your firebase stuff here
```

**Reply by mikxyas** [reply]
works like a charm. tnx

**Reply by brunocrosier** [reply]
Great! Please mark it as the answer to close the discussion

**Comment by brunocrosier** [comment]
The reason behind this is that firebase can only be initialized once, but the way you're doing it right now, Next.js tries to initialize it both on the backend and the frontend.

This method checks to see whether there it's been initialized already, and only initializes it if it has been. If not, it just returns the initialized firebase app

**Comment by sunaarun** [comment]
Check this answer , I hope it help you 

https://stackoverflow.com/questions/63492211/no-firebase-app-default-has-been-created-call-firebase-initializeapp-in

**Comment by featlast** [comment]
Hellow i need help, . I have the same problem:

`import { StyleSheet, Text, View } from 'react-native'
import React,{useState} from 'react'
import firebase from 'firebase/compat/app';

import UserLogged from './UserLogged'
import UserGuest from './UserGuest'


export default function Account() {
  const [login, setLogin] = useState(null)

  
  firebase.auth().onAuthStateChanged((user)=>{
    user !== null ? (setLogin(true)) : setLogin(false)
  })

  if (login==null){
    return <Text>Cargando...</Text>
  }
  return login ? <UserLogged/> : <UserGuest/>
}`

**Comment by david139** [comment]
Still not working, even with the recommended solution.

ERROR: FirebaseError: Firebase: No Firebase App '[DEFAULT]' has been created - call Firebase App.initializeApp() (app/no-app).

Code: 
```
const app = getApps().length < 1 ? initializeApp(firebaseConfig) : getApps()[0];
const db = getFirestore(app);
```


It works everywhere except in the getStaticProps function.

**Call stack**
getApp
.../node_modules/@firebase/app/dist/esm/index.esm2017.js (470:29)

getFirestore
.../node_modules/@firebase/firestore/dist/index.node.mjs (24870:29)

eval
webpack-internal:/lib/firestore.js (20:76)

**Comment by Dr-Raysun** [comment]
[const app = !getApps().length ? initializeApp(firebaseConfig) : getApp();](url)
